### PR TITLE
Addressing delete/create issue

### PR DESCRIPTION
**Pull request by tomikazi**

### CODE REVIEW

File: pkg/controller/proposal/controller.go [hunk 1]
```diff
@@ -509,6 +513,22 @@ func (r *Reconciler) reconcileCommit(ctx context.Context, proposal *configapi.Pr
 	}
 }
 
+func applyChangeToConfig(values map[string]*configapi.PathValue, path string, value *configapi.PathValue) (string, *configapi.PathValue) {
+	values[path] = value
```
Comment (kuujo) [inline pkg/controller/proposal/controller.go:517]: What’s the purpose of setting this key here?

File: pkg/controller/proposal/controller.go [hunk 2]
```diff
@@ -509,6 +513,22 @@ func (r *Reconciler) reconcileCommit(ctx context.Context, proposal *configapi.Pr
...
+
+	// Walk up the path and make sure that there are no parents marked as deleted in the given map, if so, remove them
+	parent := pathutils.GetParentPath(path)
+	for len(parent) > 0 {
```
Comment (kuujo) [inline pkg/controller/proposal/controller.go:521]: I think `parent != “”` would be clearer

**Review by kuujo — COMMENTED** [review]
Don’t we have the same problem with the `Apply` phase when we update the configuration’s applied values?

File: pkg/controller/proposal/controller.go (hunk 2 above)
Comment (tomikazi) [reply pkg/controller/proposal/controller.go:521]: Sure.

File: pkg/controller/proposal/controller.go (hunk 1 above)
Comment (tomikazi) [reply pkg/controller/proposal/controller.go:517]: The core of the apply change function was to update the map (as was done before, but outside this function) and then to process the path parents to clear-out any prior deletions.

**Review by kuujo — APPROVED** [review]

**Pull request merged**